Framework/Test_Framework_UnitTest.rb

```diff
@@ -0,0 +1,240 @@
+  end
+
+
+  def test_send_keys
+    create_testframe_instance = SeleniumFrameWork.new()
+    url = "file://src/Test_Framework.html"
+    create_testframe_instance.open_site(url)
+    name = "Test"
+    assert_equal("Text Sent", create_testframe_instance.send_text("id", "name", name))
+    create_testframe_instance.close_site()
+  end
+
+  def test_send_invalid_keys
+    create_testframe_instance = SeleniumFrameWork.new()
+    url = "file://src/Test_Framework.html"
+    create_testframe_instance.open_site(url)
+    name = "Test"
+    assert_equal("Text not Sent", create_testframe_instance.send_text("iddddddd", "name", name))
+    create_testframe_instance.close_site()
+  end
+
+  def test_send_keys
+    create_testframe_instance = SeleniumFrameWork.new()
+    url = "file://src/Test_Framework.html"
+    create_testframe_instance.open_site(url)
+    email = "testauto653@gmail.com"
+    assert_equal("Text Sent", create_testframe_instance.send_text("id", "email", email))
+    create_testframe_instance.close_site()
+  end
+
+  def test_send_invalid_keys
+    create_testframe_instance = SeleniumFrameWork.new()
+    url = "file://src/Test_Framework.html"
+    create_testframe_instance.open_site(url)
+    email = "testauto653@gmail.com"
+    assert_equal("Text not Sent", create_testframe_instance.send_text("class", "email", email))
+    create_testframe_instance.close_site()
+  end
+
+  def test_send_keys
+    create_testframe_instance = SeleniumFrameWork.new()
+    url = "file://src/Test_Framework.html"
+    create_testframe_instance.open_site(url)
+    pass = "1234567890"
+    assert_equal("Text Sent", create_testframe_instance.send_text("id", "password", pass))
+    create_testframe_instance.close_site()
+  end
+
+  def test_send_invalid_keys
+    create_testframe_instance = SeleniumFrameWork.new()
+    url = "file://src/Test_Framework.html"
+    create_testframe_instance.open_site(url)
+    pass = "1234567890"
+    assert_equal("Text not Sent", create_testframe_instance.send_text("css", "password", pass))
+    create_testframe_instance.close_site()
+  end
+
+  def test_send_keys
+    create_testframe_instance = SeleniumFrameWork.new()
+    url = "file://src/Test_Framework.html"
+    create_testframe_instance.open_site(url)
+    dob = "01-01-1999"
+    assert_equal("Text Sent", create_testframe_instance.send_text("id", "dob", dob))
+    create_testframe_instance.close_site()
+  end
+
+  def test_send_invalid_keys
+    create_testframe_instance = SeleniumFrameWork.new()
+    url = "file://src/Test_Framework.html"
+    create_testframe_instance.open_site(url)
+    dob = "abcd"
+    assert_equal("Text not Sent", create_testframe_instance.send_text("id", "dob", dob))
+    create_testframe_instance.close_site()
+  end
+
+  def test_scrolling
+    create_testframe_instance = SeleniumFrameWork.new()
+    url = "file://src/Test_Framework.html"
+    create_testframe_instance.open_site(url)
+    assert_equal("Scrolled", create_testframe_instance.scroll(1))
```

Comment:
This is not a good indicator of whether the page has _actually_ been scrolled.

Reply:
Modify the test html file to have some text at the particular scroll position. You can then test if this text is actually present to pass/fail the test

Reply:
Change the rest of the unit tests similarly